### PR TITLE
Support for publish to exchange

### PR DESCRIPTION
**Pull request by eriklupander**
Coulomb needed to be able to publish to a named exchange rather than to a queue (by routing key).

Added "PublishOnExchange" method to go-amqp library that accepts a single exchange name and a payload as arguments. Wrote an extremely simple unit-test.

Tested OK in coulomb integration tests. Exchange should be declared as "fanout" (which is kind of the point when not using routing keys)

### CODE REVIEW

File: amqp/rabbitmq.go [hunk 1]
```diff
@@ -162,6 +162,24 @@ func (c *RabbitMQClient) Publish(routingKey string, payload []byte) error {
...
+			Headers:         aq.Table{},
+			ContentType:     "application/json",
+			ContentEncoding: "",
+			Body:            payload,
+			DeliveryMode:    aq.Persistent,
+			Priority:        0,
+		},
+	)
+}
+
```
Comment (kristinn) [inline amqp/rabbitmq.go]: The `Publish` and `PublishOnExchange` functions are almost identical. Would be nice to make it a bit more reusable.

Comment (eriklupander) [reply amqp/rabbitmq.go]: Thanks @kristinn . Check latest commit for an updated version.

Comment (kristinn) [reply amqp/rabbitmq.go]: Good stuff! Thanks!

**Comment by eriklupander** [comment]
@cruzzan While it's true that a message in RabbitMQ always passes through an exchange, the Publish() method never specifies a named exchange. Instead, Publish() specifies the routing key and RabbitMQ then takes care of the routing by looking at what bind(s) exist for a given routing key and makes sure the message is passed through the appropriate exchange. At least for "direct" exchanges which is the opinionated way to do stuff (tm) in go-amqp.

the PublishOnExchange lets the client perform a publish to a named exchange (should be of fanout type) and does not need to know anything about consumers - it's up to the consumers to set up queues bound to the named exchange.

We could keep only a Publish(routingKey, exchangeName string) error method, which would enforce certain combinations etc and publish appropriately, but I prefer declarative methods with the intent clearly present in the method name.

**Pull request merged**